File: test/corpus/edge-case/01-invalid-colors/case.go
package main

import (
	"fmt"

	"github.com/muesli/termenv"
)

func main() {
	// Force TrueColor profile for consistent testing
	out := termenv.NewOutput(nil, termenv.WithProfile(termenv.TrueColor))
	
	fmt.Println("--- Invalid Colors Edge Case Test ---")
	
	// Test invalid hex colors
	invalidHexColors := []string{"invalid", "#ZZZZZZ", "#12345", "#1234567", "", " ", "#"}
	
	for _, invalidHex := range invalidHexColors {
		rgbColor := termenv.RGBColor(invalidHex)
		styled := out.String(fmt.Sprintf("Invalid hex '%s'", invalidHex)).Foreground(rgbColor)
		fmt.Printf("Invalid hex '%s': %s\n", invalidHex, styled)
	}
	
	// Test out-of-range ANSI colors
	invalidAnsiColors := []int{-1, 16, 256, 999}
	
	for _, invalidAnsi := range invalidAnsiColors {
		ansiColor := termenv.ANSIColor(invalidAnsi)
		styled := out.String(fmt.Sprintf("Invalid ANSI %d", invalidAnsi)).Foreground(ansiColor)
		fmt.Printf("Invalid ANSI %d: %s\n", invalidAnsi, styled)
	}
	
	// Test out-of-range ANSI256 colors
	invalidAnsi256Colors := []int{-1, 256, 300, 999}
	
	for _, invalidAnsi256 := range invalidAnsi256Colors {
		ansi256Color := termenv.ANSI256Color(invalidAnsi256)
		styled := out.String(fmt.Sprintf("Invalid ANSI256 %d", invalidAnsi256)).Foreground(ansi256Color)
		fmt.Printf("Invalid ANSI256 %d: %s\n", invalidAnsi256, styled)
	}
	
	// Test empty string styling
	emptyStyled := out.String("").Bold().Foreground(termenv.RGBColor("#FF0000"))
	fmt.Printf("Empty string styled: '%s'\n", emptyStyled)
	
	// Test whitespace-only strings
	whitespaceStyled := out.String("   ").Bold().Foreground(termenv.RGBColor("#00FF00"))
	fmt.Printf("Whitespace styled: '%s'\n", whitespaceStyled)
	
	// Test null color (NoColor equivalent)
	noColor := termenv.NoColor{}
	noColorStyled := out.String("NoColor test").Foreground(noColor)
	fmt.Printf("NoColor styled: %s\n", noColorStyled)
	
	// Test chaining with invalid colors
	chainedInvalid := out.String("Chained with invalid").
		Bold().
		Foreground(termenv.RGBColor("#INVALID")).
		Background(termenv.ANSI256Color(-5)).
		Italic()
	
	fmt.Printf("Chained with invalid: %s\n", chainedInvalid)
}

File: test/corpus/edge-case/01-invalid-colors/case.ts
import {
  ansi256Color,
  ansiColor,
  newOutput,
  noColor,
  Profile,
  rgbColor,
  withProfile,
} from '../../../../src/index.js';

// Force TrueColor profile for consistent testing
const out = newOutput(process.stdout, withProfile(Profile.TrueColor));

console.log('--- Invalid Colors Edge Case Test ---');

// Test invalid hex colors
const invalidHexColors = ['invalid', '#ZZZZZZ', '#12345', '#1234567', '', ' ', '#'];

for (const invalidHex of invalidHexColors) {
  const color = rgbColor(invalidHex);
  const styled = out.string(`Invalid hex '${invalidHex}'`).foreground(color);
  console.log(`Invalid hex '${invalidHex}': ${styled.toString()}`);
}

// Test out-of-range ANSI colors
const invalidAnsiColors = [-1, 16, 256, 999];

for (const invalidAnsi of invalidAnsiColors) {
  const color = ansiColor(invalidAnsi);
  const styled = out.string(`Invalid ANSI ${invalidAnsi}`).foreground(color);
  console.log(`Invalid ANSI ${invalidAnsi}: ${styled.toString()}`);
}

// Test out-of-range ANSI256 colors
const invalidAnsi256Colors = [-1, 256, 300, 999];

for (const invalidAnsi256 of invalidAnsi256Colors) {
  const color = ansi256Color(invalidAnsi256);
  const styled = out.string(`Invalid ANSI256 ${invalidAnsi256}`).foreground(color);
  console.log(`Invalid ANSI256 ${invalidAnsi256}: ${styled.toString()}`);
}

// Test empty string styling
const emptyStyled = out.string('').bold().foreground(rgbColor('#FF0000'));
console.log(`Empty string styled: '${emptyStyled.toString()}'`);

// Test whitespace-only strings
const whitespaceStyled = out.string('   ').bold().foreground(rgbColor('#00FF00'));
console.log(`Whitespace styled: '${whitespaceStyled.toString()}'`);

// Test null color (NoColor equivalent)
const color = noColor();
const noColorStyled = out.string('NoColor test').foreground(color);
console.log(`NoColor styled: ${noColorStyled.toString()}`);

// Test chaining with invalid colors
const chainedInvalid = out
  .string('Chained with invalid')
  .bold()
  .foreground(rgbColor('#INVALID'))
  .background(ansi256Color(-5))
  .italic();

console.log(`Chained with invalid: ${chainedInvalid.toString()}`);
